### ISSUE TITLE
[Enhancement]: A11y: Allow extensions to filter screen reader text for block cart item totals.

### ISSUE THREAD

**Issue by peterwilsoncc** [woothemes/woocommerce#62009]
### Describe the solution you'd like

The `cartItemPrice` filter exists to allow extensions to modify the pricing display to purchasers. A number of extensions use this to modify the pricing text, for example the Deposits extension uses it to change the display to `$x.xx due today`. From memory, the subscriptions extension does a similar thing.

However, this text is hidden by screen reader users in favour of descriptive text "Total price for [Product Name] item: $x.xx". Extensions are unable to modify this text to include additional information that may be required.

It would be helpful if extensions could also modify this text via a filter.

I picked this issue up while looking in to https://linear.app/a8c/issue/WOODEP-61/a11y-due-today-text-incorrectly-hidden-from-screen-readers-on-checkout

### Describe alternatives you've considered

Allowing extensions to filter whether or not the text is hidden from screen readers. As the price in the screen reader text is the value modified by an extension, I decided it was best to allow for the extension to control the full text.

### Additional context

Code reference for screen reader text:

https://github.com/woocommerce/woocommerce/blob/a3439034e3e857c352b74c6cf6e7aeae6c2d2fc6/plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/order-summary/order-summary-item.tsx#L198-L211

Code reference for displayed text:

https://github.com/woocommerce/woocommerce/blob/a3439034e3e857c352b74c6cf6e7aeae6c2d2fc6/plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/order-summary/order-summary-item.tsx#L212-L221

...and the associated filter...

https://github.com/woocommerce/woocommerce/blob/a3439034e3e857c352b74c6cf6e7aeae6c2d2fc6/plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/order-summary/order-summary-item.tsx#L113-L120

**Comment by opr** [comment]
Hi @peterwilsoncc I think this coud work:

```js
<span className="screen-reader-text">
				{ createInterpolateElement(
					sprintf(
						/* translators: %1$d is the number of items, %2$s is the item name and %3$s is the total price including the currency symbol. */
						_n(
							'Total price for %1$d %2$s item: %3$s',
							'Total price for %1$d %2$s items: %3$s',
							quantity,
							'woocommerce'
						),
						quantity,
						name,
						productPriceFormat
					),
					{
						price: (
							<>
								{ formatPrice( subtotalPrice, totalsCurrency ) }
							</>
						),
					}
				) }
			</span>
```
for example.

Do you want to try it out and make a PR?The text would be output on multiple lines like so, but I don't think that would affect anything since it's all within the `screen-reader-text` span's innerText.

<img width="244" height="80" alt="Image" src="https://github.com/user-attachments/assets/abec4fe3-258b-42bf-8e34-f6aa58a8d23b" />

**Comment by peterwilsoncc** [comment]
@opr I tried that out with Deposits and the text ends up a little contradictory with the approach as it's unclear whether it's the total or the total dues today (it's the latter): `Total price for 1 $1 Simple Product (50% deposit) item: $0.50 due today".

It looks like Deposits and Subscriptions use the same approach so the same would happen in subs too.

**Comment by opr** [comment]
@peterwilsoncc Oh, hmm yeah I see how that could be an issue, I don't think introducing a new filter specifically for the screen reader text is the right action here, we could work with the existing one, and my suggestion above, but add more information passed in context to the `cartItemPrice` filter so you can return different strings depending on where it's called?

**Comment by peterwilsoncc** [comment]
@opr Yes, I can do that. 

The filter currently has the contexts `cart` and `summary` with the latter hidden in the display and making use of screen reader text. I'll start with the name `summaryScreenReader` as the new context and put together a pull request.

**Comment by peterwilsoncc** [comment]
@opr  #62080 is ready for review and has gone through the internal processes at Fueled.
